dot/build_spec_integration_test.go

```diff
@@ -0,0 +1,165 @@
+//go:build integration
+// +build integration
+
+// Copyright 2020 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package dot
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildFromGenesis(t *testing.T) {
+	t.Parallel()
+
+	file, err := genesis.CreateTestGenesisJSONFile(false)
+	defer os.Remove(file)
+	require.NoError(t, err)
+	bs, err := BuildFromGenesis(file, 0)
+
+	expectedChainType := "TESTCHAINTYPE"
```

Comment:
nit we can use inlined constants to make it obvious to the reader a variable is not modified.

```suggestion
	const expectedChainType = "TESTCHAINTYPE"
```

Reply:
updated.